File: python/7 kyu/367.py
# https://www.codewars.com/kata/69269262ced9e95dc63abd1e/train/python

# Passed

def jumbler(indices):
    result = 0
    while indices[0] != 0:
        search_index = indices[0]
        value = indices[search_index]
        del indices[search_index]
        
        indices = [value] + indices
        result += 1
    
    return result

output = jumbler([2, 6, 8, 3, 5, 4, 0, 7, 1])
print(output)

File: nodejs/7 kyu/367.js
// https://www.codewars.com/kata/5888514674b58e929a000036/train/javascript

// Passed

function decipher(cipher) {
  let characterCodes = [];
  let currentNumberStr = String();
  for (const digit of cipher) {
    currentNumberStr += digit;
    let currentNumber = Number(currentNumberStr);
    if (currentNumber >= 97 && currentNumber <= 122) {
      characterCodes.push(currentNumber);
      currentNumberStr = String();
    }
  }

  let result = characterCodes.map((code) => String.fromCharCode(code)).join("");
  return result;
}

const output = decipher("10197115121");
console.log(output);